### PR TITLE
[CI][Codecov] Upload coverage reports once per OS only

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR uploads coverage only once in each job.

### Motivation

The first `codecov` run at the root of the repo already uploads all files. To tag each file with each module we were:
- going into every module
- uploading the module's files (that were already uploaded once) to tag it with the module and the flavor

This made a lot of codecov uploads targeting the same commit which made us reach codecov rate limit per commit:
```
We have a 100-upload limit per commit. If you are hitting the too many upload issue, that is why. I don't have a solution that works right now, but hopefully we can figure something out that will work for you.
```

### CODE REVIEW

File: tasks/libs/common/utils.py [hunk 1]
```diff
@@ -72,6 +73,22 @@ def bin_name(name):
...
 
+def get_distro():
+    """
+    Get the distro name. Windows and Darwin stays the same.
+    Linux is the only one that needs to be determined using the /etc/os-release file.
+    """
+    system = platform.system()
+    arch = platform.machine()
+    if system == 'Linux':
+        if os.path.isfile('/etc/os-release'):
```
Comment (CelianR) [inline tasks/libs/common/utils.py:84]: **🥜 nitpick:** Can you rewrite the condition with an `and` ?‏

**Review by CelianR — APPROVED** [review]

File: tasks/libs/common/utils.py [hunk 2]
```diff
@@ -72,6 +73,22 @@ def bin_name(name):
...
+            with open('/etc/os-release', encoding="utf-8") as f:
+                for line in f:
+                    if line.startswith('ID='):
+                        system = line.strip()[3:]
```
Comment (FlorentClarret) [inline tasks/libs/common/utils.py:88]: nit - I think you can break in the `if`

**Review by FlorentClarret — APPROVED** [review]

File: tasks/libs/common/utils.py (hunk 1 above)
Comment (amenasria) [reply tasks/libs/common/utils.py]: Nice catch, fixed in [278b761](https://github.com/DataDog/datadog-agent/pull/25652/commits/278b761e001f22ad3bd55f0aaf26eac07a034925), thanks !

**Pull request merged**